src/ophys_etl/qc/roi_qc.py

```diff
@@ -0,0 +1,151 @@
+                           linewidth=0.5)
+
+    binary_trace_ax.set_ylabel("Binarized F")
+    binary_trace_ax.set_xlabel("Frame Number")
+    binary_trace_ax.plot(range(len(binary_trace)), binary_trace,
+                         linewidth=0.25)
+
+    return fig
+
+
+class RoiQcReportGeneratorSchema(argschema.ArgSchema):
+    ophys_movie_path = argschema.fields.InputFile(
+        required=True,
+        description=("Path to input video (*.h5). In Allen production case, "
+                     "assumed to be motion-corrected."))
+
+    ophys_movie_h5_key = argschema.fields.Str(
+        required=False,
+        missing="data",
+        default="data",
+        description="h5py key for accessing video data array.")
+
+    suite2p_stat_path = argschema.fields.InputFile(
+        required=True,
+        description=("Path to Suite2P `stat.npy` output file which contains"
+                     "putative ROI data."))
+
+    output_dir = argschema.fields.OutputDir(
+        required=True,
+        description="Directory to save binary vs weighted ROI comparison PDF.")
+
+
+class RoiQcReportGenerator(argschema.ArgSchemaParser):
+    default_schema = RoiQcReportGeneratorSchema
+
+    def run(self):
+        movie_path = self.args["ophys_movie_path"]
+
+        # Load and convert Suite2P ROIs
+        s2p_stat = np.load(self.args["suite2p_stat_path"], allow_pickle=True)
+
+        with h5py.File(movie_path, "r") as movie_f:
+            movie_shape = movie_f[self.args["ophys_movie_h5_key"]][0].shape
+
+        weighted_rois = suite2p_rois_to_coo(s2p_stat, movie_shape)
+        binary_rois = [binarize_roi_mask(w_roi) for w_roi in weighted_rois]
+
+        # Extract traces
+        with h5py.File(movie_path, "r") as movie_f:
+            movie_frames = movie_f[self.args["ophys_movie_h5_key"]]
+
+            self.logger.info("Extracting weighted traces, please be patient.")
+            start = timer()
+            weighted_traces = extract_traces(movie_frames, weighted_rois)
+            end = timer()
+            self.logger.info(f"Weighted trace extraction: {end - start} secs")
+
+            self.logger.info("Extracting binary traces, please be patient.")
+            start = timer()
+            binary_traces = extract_traces(movie_frames, binary_rois)
+            end = timer()
+            self.logger.info(f"Binary trace extraction: {end - start} secs")
+
+        # Crop ROIs
+        cropped_weighted = [crop_roi_mask(w_roi) for w_roi in weighted_rois]
+        # Because the roi bounds can differ a bit between weighted and
+        # binarized rois, for display purposes we should binarize the cropped
+        # weighted rois.
+        cropped_binary = [binarize_roi_mask(cw_roi)
+                          for cw_roi in cropped_weighted]
+
+        # Make binary vs weighted ROI comparison plots
+        save_dir = Path(self.args["output_dir"])
+
+        # This might be a bit fragile...
+        ophys_expt_name = movie_path.split("/")[-3]
```

Comment:
For using this right now, this is fine. But, in 6 months, this might not work.
I'd suggest a priority of:
1) Use a new parameter to set this. If missing/None, then:
2) do a `re.findall("ophys_experiment_\d+", str(movie_path))` and if the length of that is not 1
3) `ophys_expt_name = "output.pdf"`
and note this in the argschema parameter description.

Reply:
Agree with both that this is unlikely to be a sustainable method, but I think it's unlikely we'll be repeating these analyses...

Reply:
@djkapner Addressed!